### PR TITLE
Milestone: Fetching data

### PR DESCRIPTION
**Pull request by Cabraham1**
### In this milestone, I achieved the following...

**I used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) for making API calls.**
**I fetched the book list from Bookstore API on the Books page load.**
**I refactored add book and remove book features to persist your changes in the server.**
**I made changes by using API.**

### CODE REVIEW

File: src/components/BookLists.js [hunk 1]
```diff
@@ -8,10 +8,10 @@ const BookLists = () => {
...
         <Book
-          key={book.id}
-          id={book.id}
-          title={book.title}
-          author={book.author}
+          key={book[0]}
+          id={book[0]}
+          title={book[1][0].title}
+          author={book[1][0].author}
         />
```
Comment (topeogunleye) [inline src/components/BookLists.js:23]: - [ ] As required by the project requirements. Kindly import useEffect (from react) and useDispatch from react-redux into your `BookList` comonent, then dispatch your fetchBook from inside useEffect.

image

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @Cabraham1, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 
> * `createAsyncThunk` is imported in the project ✔️ 
> * `fetchBooks` exists in the `booksSlice` ✔️ 
> * All HTTP requests are made with [axios](https://axios-http.com/docs/intro). ✔️ 
> * Good commit messages. :heavy_check_mark:
> * No linter errors. :heavy_check_mark:
> * Followed Github flow :heavy_check_mark:
> * Professional README file. :heavy_check_mark:
> * PR has a good title and summary :heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._
- [ ] As required, AddBook and RemoveBook should work the same as before. At the moment, `RemoveBook` does not work because your `RemoveBook` button doesn't do anything now. Kindly ensure that your RemoveBook button dispatches an action to the Redux store to remove a book

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @Cabraham1 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Cabraham1** [comment]
Thank you for reviewing and approving my code @omarsalem7

**Pull request merged**